### ISSUE TITLE
Box with desktop background color around Simdock

### ISSUE THREAD

**Issue by probonopd** [onli/simdock#9]
Running `Simdock-41df87a-x86_64.AppImage` on Xubuntu bionic Live ISO, a box with desktop background color appears around Simdock:

![](https://user-images.githubusercontent.com/2480569/82728005-77cf4800-9cdd-11ea-8304-608b9d8c176f.png)

**Comment by onli** [comment]
That's the pseudo-transparency. I tried to activate wxwidgets compositing support in the last round of fixes, as described in https://docs.wxwidgets.org/trunk/classwx_window.html#af14f8fd2ed2d30a9bbb5d4f9fd6594ec, but it didn't work. I don't think this can be improved right now, or do you have an idea what simdock could do better here?

The workaround is to use IceWMs "do not cover" flag via winoptions. [_NET_WM_STRUT_PARTIAL](https://specifications.freedesktop.org/wm-spec/wm-spec-latest.html#idm45513934529792) could be a more general alternative, reserving the space for simdock only.

**Comment by probonopd** [comment]
Unfortunately I don't know how to solve this - maybe by using Qt instead of WxWidgets?

**Comment by onli** [comment]
Maybe. I'll mark this as enhancement.

**Comment by probonopd** [comment]
Could this have been due to it running from an AppDir/AppImage?

**Comment by onli** [comment]
No, this was just how pseudo transparency works. Simdoock had no real transparency before, so it could not paint anything but the background in this situation. Not the fault of the AppImage :)

**Comment by onli** [comment]
I will close this now. The original issue was explained, and composited transparency is at hand as an alternative.
